core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java

```diff
@@ -67,14 +70,19 @@ public static ReflectiveClassBuildItem weakClass(String... className) {
 
     public static ReflectiveClassBuildItem weakClass(boolean constructors, boolean methods, boolean fields,
             String... className) {
-        return new ReflectiveClassBuildItem(constructors, methods, fields, true, false, className);
+        return new ReflectiveClassBuildItem(constructors, methods, fields, true, false, false, className);
     }
 
     public static ReflectiveClassBuildItem serializationClass(String... className) {
-        return new ReflectiveClassBuildItem(false, false, false, false, true, className);
+        return serializationClass(false, className);
     }
 
-    private ReflectiveClassBuildItem(boolean constructors, boolean methods, boolean fields, boolean weak, boolean serialization,
+    public static ReflectiveClassBuildItem serializationClass(boolean unsafeAllocated, String... className) {
+        return new ReflectiveClassBuildItem(false, false, false, false, true, unsafeAllocated, className);
+    }
+
+    public ReflectiveClassBuildItem(boolean constructors, boolean methods, boolean fields, boolean weak, boolean serialization,
+            boolean unsafeAllocated,
```

Comment:
A bit too many boolean params that will make it error prone to use. Could we please add a builder or a bit mask style constructor?

Reply:
There is a builder. And yes, I don't know why you made this constructor public. It was supposed to be private.

Could you fix that?

Reply:
> There is a builder.

I am blind, sorry for the noise!

But when it comes to the builder, could we please add parameter-less methods that set the given feature to `true`? So that less typing is needed, something like `ReflectiveClassBuildItem.builder("foo.bar.baz").methods().fields().unsafeAllocated()`?

Reply:
I'll  need a small task for tonight while waiting for builds to finish so I'll have a look at this.

Reply:
I created https://github.com/quarkusio/quarkus/pull/31222 with some enhancements.